aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java

```diff
@@ -132,14 +139,27 @@ public void configureAndInitializeInApiMode() throws JSONException, AmplifyExcep
         HubAccumulator orchestratorInitObserver =
             HubAccumulator.create(HubChannel.DATASTORE, DataStoreChannelEventName.REMOTE_SYNC_STARTED, 1)
                 .start();
+        HubAccumulator dataStoreReadyObserver =
+            HubAccumulator.create(HubChannel.DATASTORE, DataStoreChannelEventName.READY, 1)
+                .start();
+        HubAccumulator subscriptionsEstablishedObserver =
+            HubAccumulator.create(HubChannel.DATASTORE, DataStoreChannelEventName.SUBSCRIPTIONS_ESTABLISHED, 1)
+                .start();
+        HubAccumulator networkStatusObserver =
+            HubAccumulator.create(HubChannel.DATASTORE, DataStoreChannelEventName.NETWORK_STATUS, 1)
+                .start();
```

Comment:
You've got a mechanism repeated four times -- it's a signal to make it a data problem, instead:

```java
List<DataStoreChannelEventName> expectedEventNames = Arrays.asList(
    DataStoreChannelEventName.REMOTE_SYNC_STARTED,
    DataStoreChannelEventName.READY,
    DataStoreChannelEventName.SUBSCRIPTIONS_ESTABLISHED,
    DataStoreChannelEventName.NETWORK_STATUS
);
Map<DataStoreChannelEventName, HubAccumulator> accums = new HashMap<>();
for (DataStoreChannelEventName eventName : eventNames) {
    HubAccumulator accum = HubAccumulator.create(HubChannel.STORAGE, eventName, 1);
    accums.add(accum);
    accum.start();
}
```

Or actually -- for extra credit, we could make a `MultipleHubAccumulator` utility, to get some of this boilerplate out of the test class itself.

It could accept a ` LinkedHashMap` (to mandate and ordering in which to call `await()`. And have a usage pattenr like this:
```java
// It's like a list of events, in order. Allows duplicates by Integer value > 1
// HubEvent1 --> 1 time
// HubEvent2 --> 1 time
// HubEvent1 --> 5 times
// etc.
LinkedHashMap<HubEvent<?>, Integer> tape = new LinkedHashMap<>();
MultiHubAccumulator accum = MultiHubAccumulator.create(tape);

accum.start();

accum.await(); // Waits for the entire "tape" to play.
```

Reply:
I'll handle this in a separate PR :)